addon/mixins/google-object.js

```diff
@@ -251,7 +251,7 @@ var GoogleObjectMixin = Ember.Mixin.create({
   /**
    * Unlink the google object
    */
-  unlinkGoogleObject: Ember.beforeObserver('googleObject', function () {
+  unlinkGoogleObject: Ember.observer('googleObject', function () {
```

Comment:
This need to be changed so that it is called AFTER `linkGoogleObject`, so I believe both should be methods without being observers, and the `googleObject` property should be a getter/setter holding the value on `_googleObject` or so, then calling link and/or unlink depending on the old and new values